imagedephi/redact/tiff.py

```diff
@@ -100,17 +148,78 @@ def report_missing_rules(self) -> None:
 
     def report_plan(self) -> None:
         print("Tiff Metadata Redaction Plan\n")
-        for tag_value, rule in self.redaction_steps.items():
+        for tag_value, rule in self.metadata_redaction_steps.items():
             print(f"Tiff Tag {tag_value} - {rule.key_name}: {rule.action}")
         self.report_missing_rules()
-
-    def execute_plan(self) -> None:
+        print("Tiff Associated Image Redaction Plan\n")
+        print(f"Found {len(self.image_redaction_steps)} associated images")
+        if len(self.image_redaction_steps):
+            default_rule = list(self.image_redaction_steps.values())[0]
+            print(f"Redaction action: {default_rule.action}")
+
+    def create_new_image(self, ifd: IFD, rule: ImageReplaceRule, temp_dir: Path) -> Path:
+        image = None
+        if rule.replace_with == "blank_image":
+            width = int(ifd["tags"][256]["data"][0])
+            length = int(ifd["tags"][257]["data"][0])
+            output_path = temp_dir / str(uuid.uuid4())
+            image = Image.new("RGB", (width, length))
+            image.save(output_path, "TIFF", compression="jpeg")
+            return output_path
+        raise Exception("Redaction option not currently supported")
+
+    def update_new_ifd(self, old_ifd: IFD, new_ifd: IFD) -> None:
+        for tag_value, entry in old_ifd["tags"].items():
+            if tag_value not in new_ifd["tags"].keys():
```

Comment:
We probably don't want to copy all tags from the old ifd that aren't in the new ifds.  You could have tags that affect the image decoding that PIL doesn't write to the new ifd (for instance, orientation could exist on the old image but not on the new; or an icc color profile might exist on the old image).  We probably only want to copy old tags that are Datatype.ASCII.

Reply:
I'll modify the loop so it only copies over tags that are ASCII.